tlsf_cpp/example/allocator_example.cpp

```diff
@@ -0,0 +1,325 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <execinfo.h>
+#include <cxxabi.h>
+#include <malloc.h>
+
+#include <list>
+#include <memory>
+#include <string>
+
+#include "rclcpp/rclcpp.hpp"
+#include "rclcpp/strategies/allocator_memory_strategy.hpp"
+#include "std_msgs/msg/u_int32.hpp"
+#include "tlsf_cpp/tlsf.hpp"
+
+static const size_t num_rmw_tokens = 5;
+static const char * rmw_tokens[num_rmw_tokens] = {"librmw", "dds", "DDS", "dcps", "DCPS"};
+
+static bool is_running = false;
+static size_t global_runtime_deallocs = 0;
+static size_t global_runtime_news = 0;
+static size_t global_runtime_mallocs = 0;
+
+/** Check a demangled stack backtrace of the caller function for the given tokens. */
+bool check_stacktrace(const char ** tokens, size_t num_tokens, size_t max_frames = 8)
+{
+# ifndef WIN32
+  bool match = false;
+
+  // storage array for stack trace address data
+  void * addrlist[max_frames + 1];
+
+  // retrieve current stack addresses
+  int addrlen = backtrace(addrlist, sizeof(addrlist) / sizeof(void *));
```

Comment:
It would be a bit more in line with the style guide (https://google.github.io/styleguide/cppguide.html#sizeof) to say `sizeof(addrlist) / sizeof(addrlist[0])`.